src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -105,13 +105,16 @@ public String createJwt(
         List<String> roles,
         List<String> backendRoles
     ) throws Exception {
+        String tokenIdentifier = "obo";
         long timeMillis = timeProvider.getAsLong();
         Instant now = Instant.ofEpochMilli(timeProvider.getAsLong());
 
         jwtProducer.setSignatureProvider(JwsUtils.getSignatureProvider(signingKey));
         JwtClaims jwtClaims = new JwtClaims();
         JwtToken jwt = new JwtToken(jwtClaims);
 
+        jwtClaims.setProperty("typ", tokenIdentifier);
```

Comment:
This is the idea that can be also applied for the potential approach of handling the edge cases of OBO endpoint (see [comments](https://github.com/opensearch-project/security/pull/3008#discussion_r1263825789)). Since having a token type as the identifier will be easier for us to know if it is a obo token or not. So that we can have different expectations.

If we decide to go with this design, I will also update the threat model documentation